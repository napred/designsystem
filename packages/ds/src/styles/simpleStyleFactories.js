// @flow

import convertUnit from '../utilities/convertUnit';
import formatStyleResult from './formatStyleResult';
import type { Styler } from './types';
import type { System } from '../system';

export function createStyle<T: Object>(
  propName: string,
  cssAttribute: string | Array<string>,
  formatter: (val: any) => string,
  defaultValue?: any,
): Styler<T> {
  const cssAttributes = Array.isArray(cssAttribute) ? cssAttribute : [cssAttribute];

  return {
    apply: (props: Object, { viewport: bp }: System) => {
      const propValue = props[propName] == null ? defaultValue : props[propName];

      if (propValue == null) {
        return formatStyleResult(propValue, cssAttributes, formatter);
      }

      if (Array.isArray(propValue)) {
        const bpPropValue = propValue[bp];
        const lastBpPropValue = propValue[propValue.length - 1];

        if (bpPropValue != null) {
          return formatStyleResult(bpPropValue, cssAttributes, formatter);
        }

        // if propValue for bp does not exist
        // look if we have defaultValue
        // if we dont have defaultvalue, return empty result
        // if defaultValue is scalar return it
        // if defaultValue is responsive (array) pick the longer array and return last value
        if (Array.isArray(defaultValue)) {
          const bpDefaultValue = defaultValue[bp];

          if (bpDefaultValue != null) {
            return formatStyleResult(bpDefaultValue, cssAttributes, formatter);
          }

          const arr = propValue.length >= defaultValue.length ? propValue : defaultValue;
          const val = arr[arr.length - 1];

          return formatStyleResult(val, cssAttributes, formatter);
        } else if (defaultValue != null) {
          return formatStyleResult(defaultValue, cssAttributes, formatter);
        }

        // return the last breakpoint value from an array if current viewport index
        // exceeds the range
        return formatStyleResult(lastBpPropValue, cssAttributes, formatter);
      }

      return formatStyleResult(propValue, cssAttributes, formatter);
    },
    propNames: [propName],
    stripProps: [propName],
  };
}

export function createNumericStyle<T: Object>(
  propName: string,
  cssAttribute: string | Array<string>,
  defaultValue?: number | string | Array<number | string> | void,
): Styler<T> {
  return createStyle(propName, cssAttribute, convertUnit, defaultValue);
}

export function createStringStyle<T: Object>(
  propName: string,
  cssAttribute: string | Array<string>,
  defaultValue?: string | Array<string> | void,
): Styler<T> {
  return createStyle(propName, cssAttribute, val => val, defaultValue);
}
