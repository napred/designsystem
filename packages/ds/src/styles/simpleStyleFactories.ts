import { ISystem } from '../system';
import { IStyler } from '../types';
import { arrayize, convertUnit, getResponsiveValue } from '../utilities';
import formatStyleResult from './formatStyleResult';

export function createSimpleStyle<TProps extends object>(
  propName: keyof TProps,
  cssAttribute: string | string[],
  formatter: (val: any) => string | number,
  defaultValue?: string | number | null | undefined | Array<string | number | null | undefined>,
): IStyler<TProps> {
  const cssAttributes = Array.isArray(cssAttribute) ? cssAttribute : [cssAttribute];

  return {
    apply: (props: TProps, { viewport: bp }: ISystem) => {
      const propValue = arrayize<string | number | null | undefined>(props[propName] as any);
      const bpValue = getResponsiveValue(bp, propValue, defaultValue);

      return formatStyleResult(bpValue, cssAttributes, formatter);
    },
    propNames: [propName],
    stripProps: [propName],
  };
}

export function createNumericStyle<TProps extends object>(
  propName: keyof TProps,
  cssAttribute: string | string[],
  defaultValue?: string | number | null | undefined | Array<string | number | null | undefined>,
): IStyler<TProps> {
  return createSimpleStyle(propName, cssAttribute, convertUnit, defaultValue);
}

export function createStringStyle<TProps extends object>(
  propName: keyof TProps,
  cssAttribute: string | string[],
  defaultValue?: string | number | null | undefined | Array<string | number | null | undefined>,
): IStyler<TProps> {
  return createSimpleStyle(propName, cssAttribute, val => val, defaultValue);
}
