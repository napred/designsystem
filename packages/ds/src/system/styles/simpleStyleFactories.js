// @flow

import type { SystemComponentProps, StyleFn } from '../types';
import convertUnit from '../../utilities/convertUnit';
import formatStyleResult from './formatStyleResult';

export function createStyle<T: Object>(
  propName: string,
  cssAttribute: string | Array<string>,
  formatter: (val: any) => string,
  defaultValue?: any,
): StyleFn<T> {
  const cssAttributes = Array.isArray(cssAttribute) ? cssAttribute : [cssAttribute];

  const styler = (props: SystemComponentProps) => {
    const propValue = props[propName] == null ? defaultValue : props[propName];

    if (propValue == null) {
      return formatStyleResult(propValue, cssAttributes, formatter);
    }

    const bp = props.viewport;

    if (Array.isArray(propValue)) {
      const bpPropValue = propValue[bp];
      const lastBpPropValue = propValue[propValue.length - 1];

      if (bpPropValue != null) {
        return formatStyleResult(bpPropValue, cssAttributes, formatter);
      }

      // if propValue for bp does not exist
      // look if we have defaultValue
      // if we dont have defaultvalue, return empty result
      // if defaultValue is scalar return it
      // if defaultValue is responsive (array) pick the longer array and return last value
      if (Array.isArray(defaultValue)) {
        const bpDefaultValue = defaultValue[bp];

        if (bpDefaultValue != null) {
          return formatStyleResult(bpDefaultValue, cssAttributes, formatter);
        }

        const arr = propValue.length >= defaultValue.length ? propValue : defaultValue;
        const val = arr[arr.length - 1];

        return formatStyleResult(val, cssAttributes, formatter);
      } else if (defaultValue != null) {
        return formatStyleResult(defaultValue, cssAttributes, formatter);
      }

      // return the last breakpoint value from an array if current viewport index
      // exceeds the range
      return formatStyleResult(lastBpPropValue, cssAttributes, formatter);
    }

    return formatStyleResult(propValue, cssAttributes, formatter);
  };

  styler.propNames = [propName];

  return (styler: any);
}

export function createNumericStyle<T: Object>(
  propName: string,
  cssAttribute: string | Array<string>,
  defaultValue?: number | string | Array<number | string> | void,
): StyleFn<T> {
  return createStyle(propName, cssAttribute, convertUnit, defaultValue);
}

export function createStringStyle<T: Object>(
  propName: string,
  cssAttribute: string | Array<string>,
  defaultValue?: string | Array<string> | void,
): StyleFn<T> {
  return createStyle(propName, cssAttribute, val => val, defaultValue);
}
