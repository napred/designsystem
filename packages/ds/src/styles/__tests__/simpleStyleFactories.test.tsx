/**
 * @jest-environment jsdom
 */

import React from 'react';
// tslint:disable-next-line:no-implicit-dependencies
import { render } from 'react-testing-library';

import { createComponent, DesignSystem } from '../../';
import { createNumericStyle, createStringStyle, createStyle } from '../simpleStyleFactories';

const Box = createComponent('Box', 'div');

describe('simple style factories', () => {
  describe('createStyler', () => {
    it('takes the value from default responsive value if is not specified in prop (with default)', () => {
      const styles = [createStyle('test', 'test', val => val, ['1', '2'])];

      const { asFragment } = render(
        <DesignSystem is={1} styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes last value from responsive value if breakpoint exceeds the range (with default responsive)', () => {
      const styles = [createStyle('test', 'test', val => val, ['1', '2'])];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={['1']} />
          <Box test={['1', '3']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes last value from responsive value if breakpoint exceeds the range (with default simple)', () => {
      const styles = [createStyle('test', 'test', val => val, '1')];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={['1']} />
          <Box test={['1', '3']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes last value from responsive value if breakpoint exceeds the range (without default)', () => {
      const styles = [createStyle('test', 'test', val => val)];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={['1']} />
          <Box test={['1', '3']} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('createStringStyler', () => {
    it('works with simple value (without default value)', () => {
      const styles = [createStringStyle('test', 'test')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with simple value of existing css property (without default value)', () => {
      const { asFragment } = render(
        <DesignSystem>
          <Box fontWeight={600} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });


    it('works with simple value (with default value)', () => {
      const styles = [createStringStyle('test', 'test', '2')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (without default value)', () => {
      const styles = [createStringStyle('test', 'test')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default simple value)', () => {
      const styles = [createStringStyle('test', 'test', '2')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default responsive value)', () => {
      const styles = [createStringStyle('test', ['testLeft', 'testRight'], ['3', '2'])];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes the value from default responsive value if is not specified in prop (with default)', () => {
      const styles = [createStringStyle('test', ['testLeft', 'testRight'], ['3', '2'])];

      const { asFragment } = render(
        <DesignSystem is={1} styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('takes last value from responsive value if breakpoint exceeds the range (with default)', () => {
      const styles = [createStringStyle('test', ['testLeft', 'testRight'], ['3', '2'])];

      const { asFragment, rerender } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();

      rerender(
        <DesignSystem is={2} styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );
    });
  });

  describe('createNumericStyler', () => {
    it('works with simple value (without default value)', () => {
      const styles = [createNumericStyle('test', 'test')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with simple value (with default value)', () => {
      const styles = [createNumericStyle('test', 'test', '2')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={1} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (without default value)', () => {
      const styles = [createNumericStyle('test', 'test')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default simple value)', () => {
      const styles = [createNumericStyle('test', 'test', '2')];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('works with responsive value (with default responsive value)', () => {
      const styles = [createNumericStyle('test', 'test', ['1', '2'])];

      const { asFragment } = render(
        <DesignSystem styles={styles}>
          <Box test={[1]} />
          <Box />
        </DesignSystem>,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
