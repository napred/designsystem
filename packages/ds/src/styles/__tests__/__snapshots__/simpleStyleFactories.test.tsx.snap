// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple style factories createNumericStyler works with responsive value (with default responsive value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
</DocumentFragment>
`;

exports[`simple style factories createNumericStyler works with responsive value (with default simple value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
</DocumentFragment>
`;

exports[`simple style factories createNumericStyler works with responsive value (without default value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createNumericStyler works with simple value (with default value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
</DocumentFragment>
`;

exports[`simple style factories createNumericStyler works with simple value (without default value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler takes last value from responsive value if breakpoint exceeds the range (with default) 1`] = `
<DocumentFragment>
  .css-1k7vbhj {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test-left: 1px;
  test-right: 1px;
}

<div
    class="css-1k7vbhj"
  />
  .css-h64w4q {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test-left: 3px;
  test-right: 3px;
}

<div
    class="css-h64w4q"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler takes the value from default responsive value if is not specified in prop (with default) 1`] = `
<DocumentFragment>
  .css-bqqtey {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test-left: 2px;
  test-right: 2px;
}

<div
    class="css-bqqtey"
  />
  .css-bqqtey {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test-left: 2px;
  test-right: 2px;
}

<div
    class="css-bqqtey"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler works with responsive value (with default responsive value) 1`] = `
<DocumentFragment>
  .css-1k7vbhj {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test-left: 1px;
  test-right: 1px;
}

<div
    class="css-1k7vbhj"
  />
  .css-h64w4q {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test-left: 3px;
  test-right: 3px;
}

<div
    class="css-h64w4q"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler works with responsive value (with default simple value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler works with responsive value (without default value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler works with simple value (with default value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler works with simple value (without default value) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createStringStyler works with simple value of existing css property (without default value) 1`] = `
<DocumentFragment>
  .css-19jix90 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  font-weight: 600;
}

<div
    class="css-19jix90"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes last value from responsive value if breakpoint exceeds the range (with default responsive) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes last value from responsive value if breakpoint exceeds the range (with default responsive) 2`] = `
<DocumentFragment>
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
  .css-z4m3n4 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 3px;
}

<div
    class="css-z4m3n4"
  />
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes last value from responsive value if breakpoint exceeds the range (with default simple) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes last value from responsive value if breakpoint exceeds the range (with default simple) 2`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes last value from responsive value if breakpoint exceeds the range (without default) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes last value from responsive value if breakpoint exceeds the range (without default) 2`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-z4m3n4 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 3px;
}

<div
    class="css-z4m3n4"
  />
  .css-1iuax7x {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="css-1iuax7x"
  />
</DocumentFragment>
`;

exports[`simple style factories createStyler takes the value from default responsive value if is not specified in prop (with default) 1`] = `
<DocumentFragment>
  .css-1hrlp0i {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 1px;
}

<div
    class="css-1hrlp0i"
  />
  .css-a34xuh {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  test: 2px;
}

<div
    class="css-a34xuh"
  />
</DocumentFragment>
`;
